src/Test/Utilities/CoreClr/TestExecutionLoadContext.cs

```diff
@@ -0,0 +1,184 @@
+                }
+
+                return null;
+            }
+        }
+
+        private Assembly LoadPlatformAssembly(AssemblyName assemblyName, string assemblyPath)
+        {
+            lock (s_loadedPlatformAssemblies)
+            {
+                if (s_loadedPlatformAssemblies.TryGetValue(assemblyPath, out var assembly))
+                {
+                    return assembly;
+                }
+                else
+                {
+                    try
+                    {
+                        assembly = Default.LoadFromAssemblyName(assemblyName);
+                    }
+                    catch (FileNotFoundException)
+                    {
+                        // The assembly wasn't in the TPA list we can try to load
+                        // it from the dependencies list of the assembly. However,
+                        // if the assembly is mscorlib we won't be able to reload it,
+                        // no matter what
+                        if (assemblyName.Name == "mscorlib")
+                        {
+                            throw;
+                        }
+                        assembly = null;
+                    }               
+
+                    return assembly;
+                }
+            }
+        }
+
+        private Assembly LoadImageAsAssembly(ImmutableArray<byte> mainImage)
+        {
+            using (var assemblyStream = new MemoryStream(mainImage.ToArray()))
+            {
+                return LoadFromStream(assemblyStream);
+            }
+        }
+
+        internal (int ExitCode, string Output) Execute(ImmutableArray<byte> mainImage, int expectedOutputLength)
+        {
+            var mainAssembly = LoadImageAsAssembly(mainImage);
+            var entryPoint = mainAssembly.EntryPoint;
+
+            Debug.Assert(entryPoint != null, "Attempting to execute an assembly that has no entrypoint; is your test trying to execute a DLL?");
+
+            int exitCode = 0;
+            SharedConsole.CaptureOutput(() =>
+            {
+                var count = entryPoint.GetParameters().Length;
+                object[] args;
+                if (count == 0)
+                {
+                    args = Array.Empty<object>();
+                }
+                else if (count == 1)
+                {
+                    args = new[] { Array.Empty<string>() };
+                }
+                else
+                {
+                    throw new Exception("Unrecognized entry point");
+                }
+
+                exitCode = entryPoint.Invoke(null, args) is int exit ? exit : 0;
+            }, expectedOutputLength, out var stdOut, out var stdErr);
+
+            if (stdOut.Length == 0 && stdErr.Length == 0)
+            {
+            }
+
+            var output = stdOut + stdErr;
+            return (exitCode, output);
+        }
+
+        internal string DumpAssemblyData()
+        {
+            return "";
+        }
```

Comment:
Nvm, it's not part of the interface, just called from another method. I'm removing it for now. It can be implemented if we feel it's necessary.

---
In reply to: [110778296](https://github.com/dotnet/roslyn/pull/18471#discussion_r110778296) [](ancestors = 110778296,110747253)